code/headers/test_usart.hpp

```diff
@@ -0,0 +1,51 @@
+#pragma once
+#include <uart_connection.hpp>
+#include <uart_ports.hpp>
+#include <queue.hpp>
+#include <hwlib.hpp>
+
+namespace r2d2 {
+
+    class test_usart_c : public usart_connection_c {
+    private:
+        unsigned int boudrate;
+        uart_ports_c &usart_port;
+        queue_c<uint8_t, 250> input_buffer;
+
+
+    public:
+
+        test_usart_c(unsigned int baudrate, uart_ports_c controller)
```

Comment:
why is this not a reference? the variable is a reference in the class

Reply:
will fix